Tryouts/Prototypes/Shell/MainWindow.xaml.cs

```diff
@@ -40,13 +41,59 @@ public partial class MainWindow : Window
         private ManifestModel config;
         private ModuleModel[]? modules;
 
+        private string commandLineURL;
+        private string[] hereIsMyCode = (Application.Current as App).CustomerCode;
+
         public MainWindow()
         {
             InitializeComponent();
 
-            config = new ManifestParser().Manifest;
-            modules = config.Modules;
-            DataContext = modules;
+            this.Loaded += MainWindow_Loaded;
+        }
+
+        private void MainWindow_Loaded(object sender, RoutedEventArgs e)
+        {
+            if (this.hereIsMyCode.Length != 0)
+            {
+                this.LazyLoading();
+            }
+            else
+            {
+                config = new ManifestParser().Manifest;
+                modules = config.Modules;
+                DataContext = modules;
+            }
+        }
+
+        private void LazyLoading() {
+            Dictionary<string, string> commands = new Dictionary<string, string>();
+
+            this.hereIsMyCode.ToList<string>().ForEach(item => {
+                item = item.TrimStart('-');
+                string[] command = item.Split("=");
+
+                commands.Add(command[0], command[1]);
+            });
+
+            if (commands.ContainsKey("url"))
```

Comment:
```cs
if (commands.TryGetValue("url", out var url) 
{
    this.commandLineURL = url;
}
```
or
```
this.commandLineURL = commands.GetValueOrDefault("url");
```
etc. 
But you should just ditch the whole thing and use `System.CommandLine` or similar parsing library.

Reply:
First time I checked System.CommandLine, it was still in preview. Other parsing libraries have licenses that aren't good for this project, or hasn't been maintained for a while.
My idea was to switch to System.CommandLine, and rather deal with it's changes in the future. But I wanted to discuss it with the team first.

Reply:
We should be good to use System.CommandLine, that package is already being used by large projects (e.g.: dotnet CLI itself).

Reply:
That sounds reassuring.